### DISCUSSION TITLE
Ventilation system control panel

### DISCUSSION THREAD

**Discussion by MateusJK15** [FakeFishGames/Barotrauma#6877] [Ideas and suggestions]
The devs could change the way the ventilation ducts work by make them connect the atmospheres in the sub rooms, and add a ventilation control panel (similar to the image below) that allows the player to open/close the ducts, so when a duct is closed, it no longer receives oxygen from the oxygen generator and become isolated from the submarine atmosphere, that would be useful for 2 reasons :

In case of fire, it would allow the player to isolate a room in order to suffocate the fire (because as it is now, the oxygen generator continuously send oxygen through the vent).
Opening the vent after the fire is gone would reconnect the room with the submarine atmosphere, instantly replenishing it with oxygen.

This screenshot if from the game Sub Commander
![subcommander_0_28_211009_0907440](https://user-images.githubusercontent.com/86162808/136658602-7bc4334a-3f62-446c-8d18-f6170db9f226.PNG)

**Comment by SpazzMaticus107** [comment]
This honestly looks sooo cool! Imagine if we had a system that allowed us to close all doors connected to one hull by double clicking said hull on the status monitor. Overriding doors with that precision would make the captain's job so much more exciting other than staring at the sonar panel all the time. Fire in the armory but you're missing the extinguisher or crew to help you in handling that? Lockdown the room and drain oxygen by double clicking it's hull on the status monitor. Intruders inside the airlock? Delay their actions by locking down.

**Reply by ViktorShield** [reply]
You can do it without use of a terminal, but it's quite complicated and takes quite a lot of space. I see it this way:..
![image](https://user-images.githubusercontent.com/84222807/137577981-c9bb7633-1a0d-4466-8f5b-e7b4b6b55824.png)

Example: 
Quarantine | Medbay -> Closing all doors/hatches connected to the medbay "hull" if present
Flood | Reactor -> Inverting pumps in the room if present
Lights | Hull | Front -> Toggle on/off lights in the front of the sub

**Reply by SpazzMaticus107** [reply]
Oh I do know that, I just hoped this kind of technology would be available to all players and not just super creative and smart sub builders. I edit ships myself but making a complicated component grids is definitely not for me or other new players.

The way I see it is that this would be a feature of a status monitor that you can toggle just like mineral scanner in the sub editor. Then you just close the doors connected to the hull of a room you click and have them automatically force-close for 1 minute or untill you cancel that. Just how ballast flora controlls the hatches.

**Reply by mLuby** [reply]
Sounds a lot like FTL's fun door gameplay. That's a single player game though; for good co-op you wouldn't want one player sitting at the status monitor to be able to solve everything.

The sub editor can make on/off vents by linking each vent to its own hidden oxygen generator powered through a relay controlled by a switch. Though I hear your point about making it simple enough for newer sub builders to accomplish, or perhaps even in-game engineers.

I'd like the status monitor to more interactive in general: let sub builders place indicator lights, buttons, and text on it.
